.changeset/big-hats-train.md

```diff
@@ -0,0 +1,26 @@
+---
+'astro': minor
+---
+
+Adds a new optional `prerenderedErrorPageFetch` property to `RenderOptions` in the Adapter API to allow adapters to provide custom implementations for fetching prerendered error pages.
+Now, adapters can override the default `fetch()` behavior, for example when `fetch()` is unavailable or when you cannot call the server from itself.
+The following example provides a custom fetch for `500.html` and `404.html`, reading them from disk instead of performing an HTTP call:
```

Comment:
```suggestion
Adds a new optional `prerenderedErrorPageFetch` option in the Adapter API to allow adapters to provide custom implementations for fetching prerendered error pages.

Now, adapters can override the default `fetch()` behavior, for example when `fetch()` is unavailable or when you cannot call the server from itself.

The following example provides a custom fetch for `500.html` and `404.html`, reading them from disk instead of performing an HTTP call:
```
just breaking into paragraphs, and shortening the intro (maybe was too technical?)